### PR TITLE
Ms.fast test blockchain

### PR DESCRIPTION
**Pull request by mariano54**
Total time: 33m -> 13m. This includes installing and the whole CI process.

A few things contributed to the speed gains:
1. Changed test plots from k22 to k20. This saves time downloading the plots in many of the CI workflows. Also it saves time creating blocks for any tests that make blocks.
2. test_long_reorg now uses block cache and doesn't regenerate hundreds of blocks.
3. Test compact blockchain uses 2k blocks instead of 10k blocks. Testing with so many was unnecessary.   
4. A few other tests were optimized
5. Load fewer blocks if we can, for example don't load 10k blocks if you only need 1k, parsing them all is extremely slow.

Also I fixed a bug in the weight proof validation. An edge case was hit with this new list of blocks.

### CODE REVIEW

File: tests/conftest.py
Comment (arvidn) [inline tests/conftest.py:100]: this variable name is not intuitive to me. this sounds like a leftover from before the plot format was nailed down, a release candidate. Right now it just seems to be a version number in the filename.

to me it seems like this variable should be renamed to *just* `block_version` or maybe even `chain_version` and its value should be just `v1`, `v2` etc.

Comment (mariano54) [reply tests/conftest.py:100]: yeah it is just a block version, I'll change that.

**Comment by arvidn** [comment]
is this an expected flakiness? https://github.com/Chia-Network/chia-blockchain/runs/5876940304?check_suite_focus=true#step:9:83

```
_________________ TestPerformance.test_full_block_performance __________________
[77](https://github.com/Chia-Network/chia-blockchain/runs/5876940304?check_suite_focus=true#step:9:77)
tests/core/full_node/test_performance.py:174: in test_full_block_performance
[78](https://github.com/Chia-Network/chia-blockchain/runs/5876940304?check_suite_focus=true#step:9:78)
    res = await full_node_1.respond_unfinished_block(fnp.RespondUnfinishedBlock(unfinished), fake_peer)
[79](https://github.com/Chia-Network/chia-blockchain/runs/5876940304?check_suite_focus=true#step:9:79)
chia/full_node/full_node_api.py:466: in respond_unfinished_block
[80](https://github.com/Chia-Network/chia-blockchain/runs/5876940304?check_suite_focus=true#step:9:80)
    await self.full_node.respond_unfinished_block(
[81](https://github.com/Chia-Network/chia-blockchain/runs/5876940304?check_suite_focus=true#step:9:81)
chia/full_node/full_node.py:1757: in respond_unfinished_block
[82](https://github.com/Chia-Network/chia-blockchain/runs/5876940304?check_suite_focus=true#step:9:82)
    raise ConsensusError(Err(validate_result.error))
[83](https://github.com/Chia-Network/chia-blockchain/runs/5876940304?check_suite_focus=true#step:9:83)
E   chia.util.errors.ConsensusError: Error code: INVALID_CC_CHALLENGE
```

File: tests/core/full_node/test_full_node.py
Comment (arvidn) [inline tests/core/full_node/test_full_node.py:1544]: were these `4` -> `3` a consequence of changing k22 -> k20?

File: chia/full_node/weight_proof.py
Comment (arvidn) [inline chia/full_node/weight_proof.py:1510]: My understanding is that this was a (somewhat) unrelated bug fix, which I really think we ought to have a test covering as well. Do we?
I don't think a test need to hold up this PR though

**Review by arvidn — CHANGES_REQUESTED** [review]
I think this looks good. I'm pretty sure you did not mean to update the `mozilla-ca` submodule though. That should be restored before we land this.

File: chia/full_node/weight_proof.py
Comment (mariano54) [reply chia/full_node/weight_proof.py:1510]: Yeah it's a bug fix. @almogdepaz knows more about it, but it happens when there is an empty slot at the start of an epoch

**Review by arvidn — APPROVED** [review]

**Pull request merged**
